### ISSUE TITLE
problem building with iojs 3.2.0 (due to NaN changes)

### ISSUE THREAD

**Issue by amit777** [elad/node-imagemagick-native#99]
OS is centos 7.  
ImageMagick 6.9.1-10 Q16 x86_64 2015-07-27 http://www.imagemagick.org

```
  CXX(target) Release/obj.target/imagemagick/src/imagemagick.o
In file included from ../node_modules/nan/nan_new.h:190:0,
                 from ../node_modules/nan/nan.h:80,
                 from ../src/imagemagick.h:4,
                 from ../src/imagemagick.cc:9:
../node_modules/nan/nan_implementation_12_inl.h: In static member function ‘static NanIntern::FactoryBase<v8::Signature>::return_t NanIntern::Factory<v8::Signature>::New(NanIntern::Factory<v8::Signature>::FTH, int, NanIntern::Factory<v8::Signature>::FTH*)’:
../node_modules/nan/nan_implementation_12_inl.h:181:76: error: no matching function for call to ‘v8::Signature::New(v8::Isolate*, NanIntern::Factory<v8::Signature>::FTH&, int&, NanIntern::Factory<v8::Signature>::FTH*&)’
   return v8::Signature::New(v8::Isolate::GetCurrent(), receiver, argc, argv);
                                                                            ^
../node_modules/nan/nan_implementation_12_inl.h:181:76: note: candidate is:
In file included from /root/.node-gyp/3.1.0/include/node/node.h:42:0,
                 from ../src/imagemagick.h:2,
                 from ../src/imagemagick.cc:9:
/root/.node-gyp/3.1.0/include/node/v8.h:4450:27: note: static v8::Local<v8::Signature> v8::Signature::New(v8::Isolate*, v8::Handle<v8::FunctionTemplate>)
   static Local<Signature> New(
                           ^
/root/.node-gyp/3.1.0/include/node/v8.h:4450:27: note:   candidate expects 2 arguments, 4 provided
In file included from ../src/imagemagick.h:4:0,
                 from ../src/imagemagick.cc:9:
../node_modules/nan/nan.h: At global scope:
../node_modules/nan/nan.h:171:25: error: redefinition of ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Local<T>)’
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Local<T> val) {
                         ^
../node_modules/nan/nan.h:166:25: error: ‘template<class T> v8::Local<T> _NanEnsureLocal(v8::Handle<T>)’ previously declared here
 NAN_INLINE v8::Local<T> _NanEnsureLocal(v8::Handle<T> val) {
                         ^
../node_modules/nan/nan.h:564:13: error: ‘node::smalloc’ has not been declared
     , node::smalloc::FreeCallback callback
             ^
../node_modules/nan/nan.h:564:35: error: expected ‘,’ or ‘...’ before ‘callback’
     , node::smalloc::FreeCallback callback
                                   ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(char*, size_t, int)’:
../node_modules/nan/nan.h:568:50: error: ‘callback’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                  ^
../node_modules/nan/nan.h:568:60: error: ‘hint’ was not declared in this scope
         v8::Isolate::GetCurrent(), data, length, callback, hint);
                                                            ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(const char*, uint32_t)’:
../node_modules/nan/nan.h:575:67: error: call of overloaded ‘New(v8::Isolate*, const char*&, uint32_t&)’ is ambiguous
     return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
                                                                   ^
../node_modules/nan/nan.h:575:67: note: candidates are:
In file included from ../src/imagemagick.h:3:0,
                 from ../src/imagemagick.cc:9:
/root/.node-gyp/3.1.0/include/node/node_buffer.h:31:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, v8::Handle<v8::String>, node::encoding) <near match>
 NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                        ^
/root/.node-gyp/3.1.0/include/node/node_buffer.h:31:40: note:   no known conversion for argument 3 from ‘uint32_t {aka unsigned int}’ to ‘node::encoding’
/root/.node-gyp/3.1.0/include/node/node_buffer.h:43:40: note: v8::MaybeLocal<v8::Object> node::Buffer::New(v8::Isolate*, char*, size_t) <near match>
 NODE_EXTERN v8::MaybeLocal<v8::Object> New(v8::Isolate* isolate,
                                        ^
/root/.node-gyp/3.1.0/include/node/node_buffer.h:43:40: note:   no known conversion for argument 2 from ‘const char*’ to ‘char*’
In file included from ../src/imagemagick.h:4:0,
                 from ../src/imagemagick.cc:9:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(uint32_t)’:
../node_modules/nan/nan.h:579:61: error: could not convert ‘node::Buffer::New(v8::Isolate::GetCurrent(), ((size_t)size))’ from ‘v8::MaybeLocal<v8::Object>’ to ‘v8::Local<v8::Object>’
     return node::Buffer::New(v8::Isolate::GetCurrent(), size);
                                                             ^
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanBufferUse(char*, uint32_t)’:
../node_modules/nan/nan.h:586:12: error: ‘Use’ is not a member of ‘node::Buffer’
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
            ^
../src/imagemagick.cc: In function ‘void GeneratedBlobAfter(uv_work_t*)’:
../src/imagemagick.cc:450:33: warning: ‘void node::FatalException(const v8::TryCatch&)’ is deprecated (declared at /root/.node-gyp/3.1.0/include/node/node.h:282): Use FatalException(isolate, ...) [-Wdeprecated-declarations]
         FatalException(try_catch);
                                 ^
../src/imagemagick.cc: In function ‘void IdentifyAfter(uv_work_t*)’:
../src/imagemagick.cc:645:33: warning: ‘void node::FatalException(const v8::TryCatch&)’ is deprecated (declared at /root/.node-gyp/3.1.0/include/node/node.h:282): Use FatalException(isolate, ...) [-Wdeprecated-declarations]
         FatalException(try_catch);
                                 ^
In file included from ../src/imagemagick.h:4:0,
                 from ../src/imagemagick.cc:9:
../node_modules/nan/nan.h: In function ‘v8::Local<v8::Object> NanNewBufferHandle(uint32_t)’:
../node_modules/nan/nan.h:580:3: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
make: *** [Release/obj.target/imagemagick/src/imagemagick.o] Error 1
```

**Comment by horiuchi** [comment]
I try to update NaN library: #102

**Comment by elad** [comment]
@amit777 could you please try again with the master branch?

**Comment by erhhung** [comment]
The master branch works just fine (on OS X 10.11 at least) with Node 5.5, so why not publish a new NPM release with this fix? The 1.8.0 release doesn't build with any Node above 0.12.x and most folks are way past that.

**Comment by elad** [comment]
I just released the master branch as 1.9.0
